x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/recall_from_search_connectors.ts

```diff
@@ -155,12 +155,11 @@ async function recallFromLegacyConnectors({
         bool: {
           should: [
             {
-              text_expansion: {
-                [vectorField]: {
-                  model_text: text,
-                  model_id: modelId,
-                  boost,
-                },
+              sparse_vector: {
+                field: vectorField,
+                query: text,
+                inference_id: modelId,
```

Comment:
Note: This works for now, but it's kind of a bug that it does. Eventually we will want to reference an inference endpoint here and not a model ID. It's OK for now but may break some day.

Reply:
I was wondering about that. 
This is for the search connectors when adding the ML pipeline. Do you know if they create an inference endpoint behind the scenes? Afaict they just reference a model but I could be wrong.

Reply:
It's probably good to wait for the always-on endpoint to be available but I'm not sure on the timing of that. @elastic/search-extract-and-transform can help answer the connectors specific questions.

Reply:
Connectors don't create any inference endpoints behind the scenes. Through the Kibana UI for connectors, the pipelines tab where you can make an ML pipeline (not specific to connectors though), I don't believe an inference endpoint is used  - that's just directly using a model. @demjened can you confirm?

Reply:
Correct, AFAIK you can only choose models in the inference pipeline UI, not endpoints.

Reply:
Thanks all! We (Obs AI Assistant) have an (Elser) inference endpoint, `obs_ai_assistant_kb_inference`, that we are using for our own internal knowledge base. We could use that here instead of the model id. 

One caveat with that is that we'd be using Elser for inference regardless what model was used for creating the embeddings. We get around that today by applying a filter (Line 166-172) in order to only include entries for the specified model id:
```
filter: [ { term: { [modelField]: modelId, }, },],
```

Do you have any suggestion to what I should do instead when we are using an inference id?

Reply:
... actually, maybe I'm over-thinking it, and we can just keep filtering by `modelId` like in the past, while still using `sparse_vector` query with our elser inference endpoint